go-controller/pkg/util/util.go

```diff
@@ -139,3 +139,35 @@ func IsAnnotationNotSetError(err error) bool {
 	_, ok := err.(annotationNotSetError)
 	return ok
 }
+
+// CalculateHostSubnetsForClusterEntry calculates the host subnets
+// available in a CIDR entry
+func CalculateHostSubnetsForClusterEntry(cidrEntry config.CIDRNetworkEntry,
+	v4HostSubnetCount, v6HostSubnetCount *int) {
+	prefixLength, _ := cidrEntry.CIDR.Mask.Size()
+	if prefixLength > cidrEntry.HostSubnetLength {
+		klog.Warningf("Invalid cidr entry: %+v found while calculating subnet count",
```

Comment:
Nit: shouldn't we return if this happens? We will panic on the lines below otherwise with a `panic: runtime error: negative shift amount`

Reply:
But hey, maybe we should just panic if something as serious as that happens.... :shrug:

Reply:
Probably worth returning. The metrics should be the last place that causes `panic`. If there is a configuration problem, it should get caught during `config` validation in `buildKubernetesConfig`.

Reply:
Updated the PR to return.